compiler/nimconf.nim

```diff
@@ -244,6 +248,21 @@ proc getSystemConfigPath*(conf: ConfigRef; filename: RelativeFile): AbsoluteFile
     if not fileExists(result): result = p / RelativeDir"etc/nim" / filename
     if not fileExists(result): result = AbsoluteDir"/etc/nim" / filename
 
+proc configEnablesSkipParent(conf: ConfigRef; cache: IdentCache;
+                             cfgPath: AbsoluteFile): bool =
+  let prevMode = conf.skipParentDetectionMode
+  let prevSkip = optSkipParentConfigFiles in conf.globalOptions
+  conf.skipParentDetectionMode = true
```

Comment:
It seems that the order of lexer error message is changed, not sure whether that's a side effect or not. Probably, a custom empty `errorHandler` can be used for lexer